libraries/rush-lib/src/logic/operations/ShellOperationRunner.ts

```diff
@@ -370,6 +377,18 @@ export class ShellOperationRunner implements IOperationRunner {
           ensureFolderExists: true
         });
 
+        // If the operation without cache was successful, we can save the state to disk
+        if (context instanceof OperationExecutionRecord) {
```

Comment:
The point of the input context being an interface is that runners should only be interacting with that which is on the interface. The concrete class is purely meant for use by OperationExecutionManager.

Reply:
It sounds like the new `originalDurationInSeconds` field should be promoted to the interface